### ISSUE TITLE
Allow to execute "BQTestFactory.run(..)" with an explicit timeout

### ISSUE THREAD

**Issue by andrus** [bootique/bootique#200]
Per #199 we'll be unblocking a bunch of commands. So e.g. Jetty `ServerCommand` will start Jetty and exit immediately. This means we no longer need a special `JettyTestFactory` (or `BQDaemonTestFactory` for that matter) to start a Jetty app in  a test. We can run any test using standard commands with the simplest form of test factory - `BQTestFactory`, as the underlying commands will no longer be blocking. One possible downside of this approach is the inability to put a time cap on the tested stack start time. So let's address this by adding a new method `BQTestFactory.run(long timeout, TimeUnit unit)`.

**Comment by andrus** [comment]
closing, won't fix... if anyone is concerned about startup timeout, they should use BQDaemonTestRuntime
